crates/driver/src/domain/competition/solution/trade.rs

```diff
@@ -155,7 +155,75 @@ impl Trade {
                     },
                 }
             }
-            order::Kind::Limit { .. } => todo!(),
+            order::Kind::Limit { surplus_fee } => {
+                // Warning: calculating executed amounts for limit orders is complex and
+                // confusing. To understand why the calculations work, it is important to note
+                // that the solver doesn't receive limit orders with the same amounts that were
+                // specified by the users when placing the orders. Instead, the sell amount for
+                // each limit order is reduced by the surplus fee, which is the fee taken by
+                // the network to settle the order. These are referred to as "synthetic" limit
+                // orders. The surplus fees are calculated when cutting the auction.
+                //
+                // See also [`order::Kind::Limit`].
+                //
+                // Similar to market orders, the executed amounts for limit orders are
+                // calculated using the clearing prices.
+                let sell_price = clearing_prices
+                    .0
+                    .get(&sell.token)
+                    .ok_or(Error::ClearingPriceMissing)?
+                    .to_owned();
+                let buy_price = clearing_prices
+                    .0
+                    .get(&buy.token)
+                    .ok_or(Error::ClearingPriceMissing)?
+                    .to_owned();
+                match side {
+                    order::Side::Buy => Execution {
+                        buy: eth::Asset {
+                            amount: executed.into(),
+                            token: buy.token,
+                        },
+                        sell: eth::Asset {
+                            amount: executed
+                                .0
+                                .checked_mul(buy_price)
+                                .ok_or(Error::Overflow)?
+                                .checked_div(sell_price)
+                                .ok_or(Error::Overflow)?
+                                // Because of how "synthetic" limit orders are constructed as
+                                // explained above, we need to simply increase the executed sell
+                                // price by the surplus fee. We know that the user placed an order
```

Comment:
You increase the executed sell amount so you decrease the sell token price (i.e. you need more of it because its worth less).